src/Analyzers/Core/Analyzers/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs

```diff
@@ -43,253 +54,202 @@ protected AbstractUseAutoPropertyAnalyzer()
         protected abstract SyntaxNode GetFieldNode(TFieldDeclaration fieldDeclaration, TVariableDeclarator variableDeclarator);
 
         protected abstract void RegisterIneligibleFieldsAction(
-            List<AnalysisResult> analysisResults, HashSet<IFieldSymbol> ineligibleFields,
-            Compilation compilation, CancellationToken cancellationToken);
-
-        protected sealed override void InitializeWorker(AnalysisContext context)
-            => context.RegisterSemanticModelAction(AnalyzeSemanticModel);
+            ConcurrentSet<IFieldSymbol> ineligibleFields, SemanticModel semanticModel, SyntaxNode codeBlock, CancellationToken cancellationToken);
 
-        private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
-        {
-            var cancellationToken = context.CancellationToken;
-            var semanticModel = context.SemanticModel;
+        protected abstract void RegisterNonConstructorFieldWrites(
+            ConcurrentDictionary<IFieldSymbol, ConcurrentSet<SyntaxNode>> fieldWrites, SemanticModel semanticModel, SyntaxNode codeBlock, CancellationToken cancellationToken);
 
-            // Don't even bother doing the analysis if the user doesn't even want auto-props.
-            var option = context.GetAnalyzerOptions().PreferAutoProperties;
-            if (!option.Value)
```

Comment:
below is the core registration logic.  we first register for named-types, then we register for properties within that named type to find candidates to convert.  we also register for codeblocks so we can analyze all the code to see if we run into anything that would disallow the candidate from being converted.